soco/events_base.py

```diff
@@ -0,0 +1,699 @@
+                automatically shortly before timeout. Default `False`.
+        """
+
+        # TIMEOUT is provided for in the UPnP spec, but it is not clear if
+        # Sonos pays any attention to it. A timeout of 86400 secs always seems
+        # to be allocated
+        self.requested_timeout = requested_timeout
+        if self._has_been_unsubscribed:
+            raise SoCoException(
+                'Cannot resubscribe instance once unsubscribed')
+        service = self.service
+        # The Event Listener must be running, so start it if not
+        # pylint: disable=no-member
+        if not self.event_listener.is_running:
+            self.event_listener.start(service.soco)
+        # an event subscription looks like this:
+        # SUBSCRIBE publisher path HTTP/1.1
+        # HOST: publisher host:publisher port
+        # CALLBACK: <delivery URL>
+        # NT: upnp:event
+        # TIMEOUT: Second-requested subscription duration (optional)
+
+        # pylint: disable=unbalanced-tuple-unpacking
+        ip_address, port = self.event_listener.address
+        headers = {
+            'Callback': '<http://{}:{}>'.format(ip_address, port),
+            'NT': 'upnp:event'
+        }
+        if requested_timeout is not None:
+            headers["TIMEOUT"] = "Second-{}".format(requested_timeout)
+
+        # pylint: disable=missing-docstring
+        def success(headers):
+            self.sid = headers['sid']
+            timeout = headers['timeout']
+            # According to the spec, timeout can be "infinite" or "second-123"
+            # where 123 is a number of seconds.  Sonos uses "Second-123"
+            # (with a capital letter)
+            if timeout.lower() == 'infinite':
+                self.timeout = None
+            else:
+                self.timeout = int(timeout.lstrip('Second-'))
+            self._timestamp = time.time()
+            self.is_subscribed = True
+            log.info(
+                "Subscribed to %s, sid: %s",
+                service.base_url + service.event_subscription_url, self.sid)
+            # Register the subscription so it can be looked up by sid
+            # and unsubscribed at exit
+            self.subscriptions.register(self)
+
+            # Set up auto_renew
+            if not auto_renew:
+                return
+            # Autorenew just before expiry, say at 85% of self.timeout seconds
+            interval = self.timeout * 85 / 100
+            self.auto_renew_start(interval)
+
+        # pylint: disable=missing-docstring
+        def failure():
+            raise SoCoException(
+                "Could not subscribe to {}".format(
+                    service.base_url + service.event_subscription_url))
+
+        # Lock out EventNotifyHandler during registration.
+        # If events_twisted is used, this lock should always be
+        # available, since threading is not being used.
+        with self.subscriptions.subscriptions_lock:
```

Comment:
Could you please add a comment like `This is to prevent the EventNotifyHandler from sending a notification before the subscription has been registered.`?